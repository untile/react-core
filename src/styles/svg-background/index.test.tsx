/**
 * Module dependencies.
 */

import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { svgBackground } from '.';
import React from 'react';
import styled from 'styled-components';

/**
 * `Wrapper` styled component.
 */

const Wrapper = styled.div`
  ${svgBackground(
    '<svg viewBox="0 0 120 120" version="1.1" xmlns="http://www.w3.org/2000/svg"><circle cx="60" cy="60" r="50"/></svg>'
  )}
`;

/**
 * Test `svgBackground` style util.
 */

describe(`'svgBackground' style util`, () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Wrapper />);

    expect(asFragment()).toMatchSnapshot();
  });
});
