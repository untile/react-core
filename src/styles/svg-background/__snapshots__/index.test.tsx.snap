// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'svgBackground' style util matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-image: url('data:image/svg+xml;utf8,%3Csvg%20viewBox%3D%220%200%20120%20120%22%20version%3D%221.1%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Ccircle%20cx%3D%2260%22%20cy%3D%2260%22%20r%3D%2250%22%2F%3E%3C%2Fsvg%3E');
}

<div
    class="c0"
  />
</DocumentFragment>
`;
